### PR TITLE
Move attributes required for SRV record creation into the necessary data field

### PR DESCRIPTION
**Pull request by AceFire6**
The values the SRV record uses are required to be held in `data` as defined [here](https://api.cloudflare.com/#dns-records-for-a-zone-properties): 

> SRV Record
> ```
{
  "type": "SRV",
  ...,
  "data": {
    "service": "_sip",
    "proto": "_tcp",
    "name": "example.com",
    "priority": 10,
    "weight": 5,
    "port": 8806,
    "target": "somewhere.com"
  }
}
```

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by AceFire6** [comment]
I'm not really sure what to do with tests. They no longer work because the `Record`'s `data` entry is an empty dictionary instead of what is returned normally.

**Comment by stefanor** [comment]
Do we know why data is empty?

Record's dirty data handling also doesn't deal with dictionary attributes. It assumes all the attributes are immutable.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: tests/fakes.py [hunk 1]
```diff
@@ -160,8 +160,9 @@ def create_dns_record(self, zone_id, data):
             'zone_name': self.zones[zone_id]['name'],
             'created_on': '2014-01-01T05:20:00.12345Z',
             'modified_on': '2014-01-01T05:20:00.12345Z',
-            'data': {},
         })
+        if 'data' not in data:
+            data['data'] = {}
```
Comment (stefanor) [inline tests/fakes.py]:
```python
data.setdefault('data', {})
```

**Comment by stefanor** [comment]
Throw a changelog entry in, too.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by stefanor** [comment]
We still need to deal with:

> Record's dirty data handling also doesn't deal with dictionary attributes. It assumes all the attributes are immutable.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: pycloudflare/models.py [hunk 2]
```diff
@@ -191,36 +192,40 @@ def __repr__(self):
 
 class Record(object):
     _data = ()
+    _record_attrs = ('zone', '_service', '_data',
```
Comment (RayeN) [inline pycloudflare/models.py]: `_own_attrs`?

File: pycloudflare/models.py [hunk 3]
```diff
@@ -191,36 +192,40 @@ def __repr__(self):
...
             return self._data[name]
         raise AttributeError()
 
     def __setattr__(self, name, value):
-        if name in ('zone', '_service', '_data', '_updates'):
+        if name in self._record_attrs:
             return super(Record, self).__setattr__(name, value)
         if name in self._data:
-            self._updates[name] = value
+            self._clear_cache = (name == 'name')
```
Comment (RayeN) [inline pycloudflare/models.py]: I would use `if` here

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: pycloudflare/models.py [hunk 4]
```diff
@@ -191,36 +192,41 @@ def __repr__(self):
...
         raise AttributeError()
 
     def __setattr__(self, name, value):
-        if name in ('zone', '_service', '_data', '_updates'):
+        if name in self._own_attrs:
             return super(Record, self).__setattr__(name, value)
         if name in self._data:
-            self._updates[name] = value
+            if name == 'name':
+                self._clear_cache = True
```
Comment (stefanor) [inline pycloudflare/models.py]: What's the reason for this check?

**Comment by stefanor** [comment]
I don't see how these changes actually help. `Record.data` is still mutable, and `Record` is unaware of mutation.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by AceFire6** [comment]
@stefanor Yeah, initially I misunderstood exactly what you were talking about. I think this newest change should address that as it accounts for the full `Record._data` `dict`.

File: pycloudflare/models.py [hunk 5]
```diff
@@ -191,36 +194,36 @@ def __repr__(self):
...
-        self._updates = {}
+        self._old_data = deepcopy(data)
 
     def __getattr__(self, name):
-        if name in self._updates:
-            return self._updates[name]
         if name in self._data:
             return self._data[name]
+        if name in self._old_data:
+            return self._old_data[name]
```
Comment (stefanor) [inline pycloudflare/models.py]: Why do we return anything from `_old_data`? `_data` is initialized with everything.

File: pycloudflare/models.py [hunk 6]
```diff
@@ -191,36 +194,36 @@ def __repr__(self):
...
 class Record(object):
     _data = ()
+    _own_attrs = ('zone', '_service', '_data', '_old_data')
 
     def __init__(self, zone, data):
         self.zone = zone
         self._service = zone._service
         self._data = data
-        self._updates = {}
+        self._old_data = deepcopy(data)
```
Comment (stefanor) [inline pycloudflare/models.py]: I'd be tempted to call that `_init_data` or `_saved_data`. "old" doesn't quite capture what's going on.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by stefanor** [comment]
:+1:

**Pull request merged**